x-pack/plugins/screenshotting/README.md

```diff
@@ -3,18 +3,153 @@
+The intitalization process downloads (if it is not already present) and verifies the Chromium build.
 
-## Testing Chromium downloads
+The start contract exposes a public API to interact with the plugin.
+Apart from the actual screenshotting functionality, it also provides a way for self-diagnostics.
 
-To download all Chromium browsers for all platforms and architectures:
+Here is an example of how you can take a screenshot of a Kibana URL.
+
+```typescript
+import type { CoreSetup, Plugin } from 'src/core/server';
+import type { ScreenshottingStart } from 'x-pack/plugins/screenshotting/server';
+
+
+interface StartDeps {
+  screenshotting: ScreenshottingStart;
+}
+
+class ExamplePlugin implements Plugin<void, void, void, StartDeps> {
+  setup({ http, getStartServices }: CoreSetup<StartDeps>) {
+    const router = http.createRouter();
+
+    router.get(
+      {
+        path: '/api/capture',
+        validate: {
+          query: schema.object({
+            id: schema.string(),
+          }),
+        },
+      },
+      async (context, request, response) => {
+        const [, { screenshotting }] = await getStartServices();
+        const { metrics, results } = await screenshotting
+          .getScreenshots({
+            request,
+            urls: [`http://localhost/app/canvas#/workpad/workpad-${request.query.id}`],
+          })
+          .toPromise();
+
+        return response.ok({
+          body: JSON.stringify({
+            metrics,
+            image: results[0]?.screenshots[0]?.data.toString('base64'),
+            errors: results[0]?.renderErrors,
+          } as ScreenshottingExpressionResponse),
+        });
+      }
+    );
+  }
+
+  start() {}
+}
+
+export function plugin() {
+  return new ExamplePlugin();
+}
+```
+
+### API
+Please use automatically generated API reference or generated TypeDoc comments to find the complete documentation.
+
+#### `getScreenshots(options): Observable`
+Takes screenshots of multiple pages or an expression and returns an observable with the screenshotting results.
 
+The `options` parameter is an object with parameters of the screenshotting session.
+Option | Required | Default | Description
+--- | :---: | --- | ---
+`browserTimezone` | no | _none_ | The browser timezone that will be emulated in the browser instance. This option should be used to keep timezone on server and client in sync.
+`expression` | no | _none_ | An expression to capture screenshot of. Mutually exclusive with the `urls` parameter.
+`format` | no | `'png'` | An output format. It can either be PDF or PNG. In case of capturing multiple URLs, all the screenshots will be combined into one document for PDF format. For PNG format, an array of screenshots will be returned.
+`headers` | no | _none_ | Custom headers to be sent with each request. The headers will be used for authorization.
+`input` | no | `undefined` | The expression input.
```

Comment:
Is `input` also mutually exclusive of `urls`? Is it required if `expression` is provided?

Reply:
No, it is optional for the `expression` too.